src/main/java/net/starschema/clouddb/jdbc/BQPreparedStatement.java

```diff
@@ -306,6 +306,11 @@ public int executeUpdate() throws SQLException {
         throw new BQSQLFeatureNotSupportedException("executeUpdate()");
     }
 
+    @Override
+    public ResultSet executeQuery(String querySql) throws SQLException {
+        throw new BQSQLFeatureNotSupportedException("executeQuery(String querySQL)");
+    }
+
```

Comment:
I take it this is there to implement the Statement Interface? By doing this, the default way to run queries with a string (part of the JDBC spec) would be unsupported. I'm not sure if anyone actually uses this with prepared statements. 

It looks like this problem is originally from a change that was made to 
https://github.com/jonathanswenson/starschema-bigquery-jdbc/blob/275a459c56058191206bcaced99f0799e893f1ca/src/main/java/net/starschema/clouddb/jdbc/BQStatementRoot.java#L243

Here we added the functionality to pass in unlimitedBillingBytes. So what probably happened is that we had a cached version of BQPreparedStatement.class around so it didn't recompile because it didn't realize it had to when the base class changed. 

Would it make sense to just add an executeQuery method to BQStatementRoot that has a default value (such as unlimitedBillingBytes set to false instead)?

Reply:
Ah, yeah it looks like we just tweaked the basic implementation in the BQStatement class in https://github.com/jonathanswenson/starschema-bigquery-jdbc/pull/33/files

Reply:
I was toying with the idea of adding a default-valued `executeQuery(String querySql)` to the root statement - however, there is a comment in the base definition of `Statement.java` for `executeQuery(String querySql)` that says "Note: This method cannot be called on a PreparedStatement or CallableStatement." 

I think it's `executeQuery()` (no parameters) that's getting called for PreparedStatements - is there actually a use case for `executeQuery(String querySql)` that I'm missing?

Reply:
After all, a prepared statement already has its SQL defined, so calling for the execution of a query with different SQL is a bit weird

Reply:
Ah but I see what you mean, it's a bit weird to override a function that shouldn't exist on this interface - it should really exist on the base Statement class. I'll make that change.

Reply:
Technically it MUST be there for both. That's part of the statement interface even though it doesn't really make sense...

Reply:
https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#executeQuery(java.lang.String)

Reply:
Though because of what you said, it might be ok to just have an unimplemented error. 

What was the behavior before we added the billingbytes flag?

Reply:
i.e. calling `PreparedStatement.executeQuery(sql)` pre-billingbytes change? Hmm, I can try to figure that out. Also note that even the docs you sent indicate that "Note:This method cannot be called on a PreparedStatement or CallableStatement." so maybe it does make sense to keep this unimplemented? If someone is trying to call `PreparedStatement.executeQuery(sql)`, they're doing something wrong in the first place...

Reply:
Missed that.... great. Assuming that this BQFeaturNotSupported exception is a SQL Exception then I think that is fine.

Maybe it deserves a different exception type/name since isn’t supported by the spec  (but saying it isn’t supported is probably ok).